File: Medium/56. Merge Intervals/solution.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        vector<vector<int> > result;
        
        int preLeft = intervals[0][0];
        int preRight = intervals[0][1];
        for (int i = 1; i < intervals.size(); i++){
            if (preLeft <= intervals[i][0] && intervals[i][0] <= preRight){
                preRight = max(preRight, intervals[i][1]);
            }
            else{
                result.push_back(vector<int> {preLeft, preRight});
                preLeft = intervals[i][0];
                preRight = intervals[i][1];
            }
        }
        result.push_back(vector<int> {preLeft, preRight});
        return result;
    }
};

File: Medium/56. Merge Intervals/solution.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        ret = []

        l = intervals[0][0]
        r = intervals[0][1]

        for i in range(1, len(intervals)):
            if r < intervals[i][0]:
                ret.append([l, r])
                l = intervals[i][0]
                r = intervals[i][1]

            else:
                r = max(intervals[i][1], r)
        
        ret.append([l, r])

        return ret